From 10d8c8b873fbd4da2bd258f237b59b6efb15ae8d Mon Sep 17 00:00:00 2001
From: George Cherian <george.cherian@marvell.com>
Date: Mon, 24 May 2021 11:53:50 +0530
Subject: [PATCH 1540/1921] octeontx2-af: Fix the KPU support to parse PPPoE
 packets

The KPU support for the following
 1. L2 + PPPoE + PPP + IPv4
 2. L2 + PPPoE + PPP + IPv6

flows were dropped by the default mkex as the DMAC extraction
failed.

Fixes: 18f037e9cbfbd7 ("octeontx2-af: Add KPU support to parse PPPoE
packets")
Signed-off-by: George Cherian <george.cherian@marvell.com>

Change-Id: Iefe8378fcd2c069e3ed298397b5af40b1a31e503
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/52482
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <sgoutham@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 .../net/ethernet/marvell/octeontx2/af/npc_profile.h  | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h b/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h
index ceffaba3b493..f8db5bb7cead 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h
+++ b/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h
@@ -8727,9 +8727,9 @@ static struct npc_kpu_profile_action kpu1_action_entries[] = {
 	},
 	{
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
-		0, 0, 0, 0, 1,
-		NPC_S_NA, 0, 1,
-		NPC_LID_LA, NPC_LT_LA_8023,
+		8, 12, 0, 2, 0,
+		NPC_S_KPU4_PPPOE, 12, 1,
+		NPC_LID_LA, NPC_LT_LA_ETHER,
 		0,
 		0, 0, 0, 0,
 	},
@@ -8743,9 +8743,9 @@ static struct npc_kpu_profile_action kpu1_action_entries[] = {
 	},
 	{
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
-		8, 12, 0, 2, 0,
-		NPC_S_KPU4_PPPOE, 12, 1,
-		NPC_LID_LB, NPC_LT_LB_PPPOE,
+		0, 0, 0, 0, 1,
+		NPC_S_NA, 0, 1,
+		NPC_LID_LA, NPC_LT_LA_8023,
 		0,
 		0, 0, 0, 0,
 	},
-- 
2.31.1

